File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:

        while len(tokens) > 1:
            for i in range(len(tokens)):
                if tokens[i] in '+-*/':
                    a, b = int(tokens[i-2]), int(tokens[i-1])
                    if tokens[i] == '+':
                        ans = a+b
                    elif tokens[i] == '-':
                        ans = a-b
                    elif tokens[i] == '*':
                        ans = a*b
                    else:
                        ans = int(a/b)
                    tokens = tokens[:i-2] + [str(ans)] + tokens[i+1:]
                    break
        return int(tokens[0])

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
#include <vector>
#include <unordered_map>
#include <algorithm>
using namespace std;

class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> cnt;

        for (int i : nums) {
            cnt[i]++;
        }

        vector<pair<int,int>> arr;

        for (const auto& p : cnt) {
            arr.push_back({p.second, p.first});
        }

        sort(arr.begin(), arr.end());

        vector<int> ans;

        for (int i = arr.size() - 1; k > 0; i--, k--) {
            ans.push_back(arr[i].second);
        }

        return ans;
    }
};